open-api/rest-catalog-open-api.yaml

```diff
@@ -3405,6 +3490,101 @@ components:
       allOf:
         - $ref: '#/components/schemas/ScanTasks'
 
+    GetEventsRequest:
+      type: object
+      properties:
+        page-token:
+          $ref: "#/components/schemas/PageToken"
+        page-size:
+          type: integer
+          format: int32
+          description: >
+            The maximum number of events to return in a single response.
+            If not provided, the server may choose a default page size.
+            Servers may return less results than requested for various reasons, such as
+            server side limits, payload size or processing time.
+        after-timestamp-ms:
+          type: integer
+          format: int64
+          description: >
+            The (server) timestamp in milliseconds to start consuming events from (inclusive).
+            If not provided, the first available timestamp is used.
+        operation-types:
+          type: array
+          items:
+            $ref: "#/components/schemas/OperationType"
+          description: >
+            Filter events by the type of operation.
+            If not provided, all types are returned.
+        catalog-objects:
+          type: array
+          discriminator:
+            propertyName: reference-type
+            mapping:
+              namespace: "#/components/schemas/NamespaceReference"
+              table: "#/components/schemas/TableReference"
+              view: "#/components/schemas/ViewReference"
+          items:
+            oneOf:
+              - $ref: "#/components/schemas/NamespaceReference"
+              - $ref: "#/components/schemas/TableReference"
+              - $ref: "#/components/schemas/ViewReference"
+          description: >
+            List of catalog objects (namespaces, tables, views) to get events for.
+            If not provided, events for all objects will be returned subject to other filters.
+            For specified namespaces, events for the namespaces and all containing objects
+            (namespaces, tables, views) will be returned.
```

Comment:
Minor: is "will" correct if this is defining what services should or must do? I would expect this to be "must be returned" if we want to specify the service behavior.

In addition, is the requirement to return changes for child objects recursive or for one layer only?

Reply:
Changed to "must". Added hint for recursive.

Reply:
After discussing this we ended up keeping it recursive for the following reasons:
- Otherwise not possible to sub-select an area of the Catalog without previously crawling it
- Otherwise not possible to efficiently listen to Create Events, as specific identifiers are not yet known
- Consistent with the behaviour on Warehouse level which is recursive.